tinygrad/tensor.py

```diff
@@ -341,10 +341,9 @@ def repeat(self, repeats):
 
   # TODO: make this nicer with syntactic sugar in slice
   def chunk(self, num, dim):
-    slice_params = [[(0, s) for s in self.shape] for _ in range(num)]
-    for i,k in enumerate(range(0, self.shape[dim], self.shape[dim]//num)):
```

Comment:
changing
```python
self.shape[dim]//num -> ceil(self.shape[dim]/num)
```  
in all occurences fixes the bug in prev code, it was late for me to realize this unfortunately.

Reply:
I think the old chunk code is more readable, maybe try the fix on the old code and see if it passes the test?

Reply:
Hello @wozeparrot,

Most of it looking the same only using slice(). But if still unreadable than I would directly update the old chunk code.

I am insisting on not using tensor.slice but instead using builtin slice(). Sometimes tensor.slice() can add unexpected zero padding.

Reply:
It's much more readable now. :)